jgrapht-core/src/main/java/org/jgrapht/alg/tour/HeldKarpTSP.java

```diff
@@ -112,6 +103,10 @@ private double memo(int previousNode, int state, double[][] C, double[][] W){
             throw new IllegalArgumentException("Graph contains no vertices");
         }
 
+        if (n > 31){
+            throw new IllegalArgumentException("Graph contains more than 31 vertices");
```

Comment:
This is a really weird error message. Solving a TSP on a graph which has more than 31 vertices, is very common. Please make this error more descriptive, e.g.:
"The internal representation of the dynamic programming state space cannot represent graphs containing more than 31 vertices. The runtime complexity of this class (O(2^|V| x |V|^2)) makes it unsuitable for graphs with more than 31 vertices."

Reply:
Updated.

Reply:
I don't think you need the quotes, (i.e. remove `\"`).